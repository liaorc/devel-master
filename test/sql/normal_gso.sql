--#
--#       Gpu Sort Simple TestCases. 
--#

set pg_strom.debug_force_gpusort to on;
set pg_strom.gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set pg_strom.enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;

select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;

select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;

