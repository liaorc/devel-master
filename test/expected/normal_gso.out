--#
--#       Gpu Sort Simple TestCases. 
--#
set pg_strom.debug_force_gpusort to on;
set pg_strom.gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set pg_strom.enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;
select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |       31
  2000 |       28
  3000 |       24
  4000 |       21
  5000 |       17
  6000 |       14
  7000 |       10
  8000 |        7
  9000 |        3
 10000 |        0
(10 rows)

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |        3
  2000 |        7
  3000 |       10
  4000 |       14
  5000 |       17
  6000 |       21
  7000 |       24
  8000 |       28
  9000 |       31
 10000 |         
(10 rows)

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |       -2
  2000 |       -5
  3000 |       -9
  4000 |      -12
  5000 |      -16
  6000 |      -19
  7000 |      -23
  8000 |      -26
  9000 |      -29
 10000 |      -33
(10 rows)

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |      -29
  2000 |      -26
  3000 |      -23
  4000 |      -19
  5000 |      -16
  6000 |      -12
  7000 |       -9
  8000 |       -5
  9000 |       -2
 10000 |         
(10 rows)

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |       29
  2000 |       23
  3000 |       16
  4000 |        9
  5000 |        2
  6000 |       -5
  7000 |      -12
  8000 |      -18
  9000 |      -26
 10000 |      -33
(10 rows)

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |      -26
  2000 |      -18
  3000 |      -12
  4000 |       -5
  5000 |        2
  6000 |        9
  7000 |       16
  8000 |       23
  9000 |       29
 10000 |         
(10 rows)

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |         
  2000 |         
  3000 |         
  4000 |         
  5000 |         
  6000 |         
  7000 |         
  8000 |         
  9000 |         
 10000 |         
(10 rows)

select rowid,smlint_x from (select smlint_x, row_number() over (order by smlint_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | smlint_x 
-------+----------
  1000 |         
  2000 |         
  3000 |         
  4000 |         
  5000 |         
  6000 |         
  7000 |         
  8000 |         
  9000 |         
 10000 |         
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |   2033457
  2000 |   1812091
  3000 |   1585141
  4000 |   1365225
  5000 |   1146124
  6000 |    916219
  7000 |    692177
  8000 |    462166
  9000 |    234334
 10000 |        28
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |    234063
  2000 |    462003
  3000 |    691846
  4000 |    916191
  5000 |   1145778
  6000 |   1365173
  7000 |   1585071
  8000 |   1812018
  9000 |   2033443
 10000 |          
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |   -108476
  2000 |   -338431
  3000 |   -560655
  4000 |   -782385
  5000 |  -1011155
  6000 |  -1236890
  7000 |  -1468465
  8000 |  -1700227
  9000 |  -1928925
 10000 |  -2147350
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |  -1929012
  2000 |  -1700243
  3000 |  -1468605
  4000 |  -1237002
  5000 |  -1011155
  6000 |   -782459
  7000 |   -560841
  8000 |   -338525
  9000 |   -108528
 10000 |          
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |   1918639
  2000 |   1452304
  3000 |    994445
  4000 |    555661
  5000 |    112481
  6000 |   -341392
  7000 |   -785029
  8000 |  -1250819
  9000 |  -1697691
 10000 |  -2145725
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |  -1698068
  2000 |  -1250880
  3000 |   -785728
  4000 |   -341596
  5000 |    112333
  6000 |    554682
  7000 |    993924
  8000 |   1450731
  9000 |   1918573
 10000 |          
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |          
  2000 |          
  3000 |          
  4000 |          
  5000 |          
  6000 |          
  7000 |          
  8000 |          
  9000 |          
 10000 |          
(10 rows)

select rowid,integer_x from (select integer_x, row_number() over (order by integer_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | integer_x 
-------+-----------
  1000 |          
  2000 |          
  3000 |          
  4000 |          
  5000 |          
  6000 |          
  7000 |          
  8000 |          
  9000 |          
 10000 |          
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid |     bigint_x     
-------+------------------
  1000 | 8752238854905266
  2000 | 7787119966456644
  3000 | 6794098432127533
  4000 | 5852154899445318
  5000 | 4837065290573414
  6000 | 3855970433748697
  7000 | 2850882639773040
  8000 | 1915735763987726
  9000 |  961652961509900
 10000 |     398590144938
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid |     bigint_x     
-------+------------------
  1000 |  961600425469936
  2000 | 1913781025587069
  3000 | 2848860431731130
  4000 | 3853544760773968
  5000 | 4835695805891346
  6000 | 5850396591373943
  7000 | 6793871211177705
  8000 | 7783096681611919
  9000 | 8750191310851277
 10000 |                 
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid |     bigint_x      
-------+-------------------
  1000 |  -483943634874401
  2000 | -1453136089349882
  3000 | -2386446437119426
  4000 | -3378863781219140
  5000 | -4355433954422030
  6000 | -5354325704022426
  7000 | -6302129612205851
  8000 | -7289545067457413
  9000 | -8277034542175355
 10000 | -9221992927150998
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid |     bigint_x      
-------+-------------------
  1000 | -8277473397638693
  2000 | -7289759429275156
  3000 | -6302226850265432
  4000 | -5355124198572294
  5000 | -4356051690978280
  6000 | -3379089898362372
  7000 | -2386648725784101
  8000 | -1455015837621551
  9000 |  -484275622961480
 10000 |                  
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid |     bigint_x      
-------+-------------------
  1000 |  8182981080584815
  2000 |  6297875292480340
  3000 |  4268877268052345
  4000 |  2353642189877150
  5000 |   461941659878294
  6000 | -1441615075801563
  7000 | -3369743173352948
  8000 | -5328866452151730
  9000 | -7298305675459822
 10000 | -9219276063983534
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid |     bigint_x      
-------+-------------------
  1000 | -7303673628665577
  2000 | -5328979366841942
  3000 | -3373591438280360
  4000 | -1443709404934308
  5000 |   457790745785401
  6000 |  2353201199815066
  7000 |  4268869245053436
  8000 |  6295958912432603
  9000 |  8181616517935268
 10000 |                  
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | bigint_x 
-------+----------
  1000 |         
  2000 |         
  3000 |         
  4000 |         
  5000 |         
  6000 |         
  7000 |         
  8000 |         
  9000 |         
 10000 |         
(10 rows)

select rowid,bigint_x from (select bigint_x, row_number() over (order by bigint_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | bigint_x 
-------+----------
  1000 |         
  2000 |         
  3000 |         
  4000 |         
  5000 |         
  6000 |         
  7000 |         
  8000 |         
  9000 |         
 10000 |         
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid | real_x 
-------+--------
  1000 | 0.9439
  2000 | 0.8393
  3000 | 0.7312
  4000 |  0.628
  5000 | 0.5249
  6000 | 0.4204
  7000 | 0.3161
  8000 | 0.2101
  9000 | 0.1015
 10000 | 0.0003
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid | real_x 
-------+--------
  1000 | 0.1015
  2000 |   0.21
  3000 | 0.3158
  4000 | 0.4204
  5000 | 0.5248
  6000 |  0.628
  7000 | 0.7304
  8000 | 0.8392
  9000 | 0.9438
 10000 |       
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid | real_x  
-------+---------
  1000 | -0.0481
  2000 | -0.1567
  3000 | -0.2595
  4000 | -0.3675
  5000 | -0.4686
  6000 |  -0.578
  7000 |  -0.686
  8000 |  -0.787
  9000 | -0.8883
 10000 | -0.9998
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid | real_x  
-------+---------
  1000 | -0.8884
  2000 | -0.7871
  3000 |  -0.686
  4000 |  -0.578
  5000 | -0.4687
  6000 | -0.3675
  7000 | -0.2595
  8000 | -0.1569
  9000 | -0.0484
 10000 |        
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid | real_x  
-------+---------
  1000 |  0.8894
  2000 |  0.6784
  3000 |  0.4681
  4000 |  0.2565
  5000 |  0.0543
  6000 | -0.1502
  7000 | -0.3595
  8000 | -0.5701
  9000 | -0.7849
 10000 | -0.9999
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid | real_x  
-------+---------
  1000 | -0.7853
  2000 | -0.5709
  3000 | -0.3596
  4000 | -0.1503
  5000 |   0.054
  6000 |  0.2564
  7000 |  0.4678
  8000 |  0.6777
  9000 |  0.8894
 10000 |        
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | real_x 
-------+--------
  1000 |       
  2000 |       
  3000 |       
  4000 |       
  5000 |       
  6000 |       
  7000 |       
  8000 |       
  9000 |       
 10000 |       
(10 rows)

select rowid,real_x from (select real_x, row_number() over (order by real_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | real_x 
-------+--------
  1000 |       
  2000 |       
  3000 |       
  4000 |       
  5000 |       
  6000 |       
  7000 |       
  8000 |       
  9000 |       
 10000 |       
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid |     float_x     
-------+-----------------
  1000 | 0.9481582744047
  2000 | 0.8412061007693
  3000 | 0.7362818289548
  4000 | 0.6291552940384
  5000 | 0.5267555210739
  6000 | 0.4145472459495
  7000 | 0.3109927624464
  8000 | 0.2085867277347
  9000 | 0.1043348135427
 10000 | 0.0003206986003
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid |     float_x     
-------+-----------------
  1000 | 0.1040899110958
  2000 | 0.2084929337725
  3000 | 0.3105730316602
  4000 | 0.4145092964172
  5000 | 0.5266925939359
  6000 | 0.6289276690222
  7000 | 0.7362595382147
  8000 | 0.8406651820987
  9000 | 0.9481558552943
 10000 |                
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid |     float_x      
-------+------------------
  1000 | -0.0522141857073
  2000 | -0.1568604102358
  3000 |  -0.260433471296
  4000 | -0.3693278222345
  5000 | -0.4790044301189
  6000 |  -0.585879619699
  7000 | -0.6871913410723
  8000 | -0.7886257874779
  9000 | -0.8959867861122
 10000 |  -0.999974144157
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid |     float_x      
-------+------------------
  1000 | -0.8959875097498
  2000 | -0.7890867833048
  3000 | -0.6872865371406
  4000 | -0.5859081516974
  5000 | -0.4790461650118
  6000 | -0.3694010148756
  7000 | -0.2605017554015
  8000 | -0.1568937595002
  9000 | -0.0522781433538
 10000 |                 
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid |     float_x      
-------+------------------
  1000 |  0.9009603494778
  2000 |  0.6908911308274
  3000 |  0.4926429335028
  4000 |  0.2820586962625
  5000 |  0.0664466973394
  6000 | -0.1519305901602
  7000 | -0.3694218369201
  8000 | -0.5868804641068
  9000 | -0.7999544451013
 10000 | -0.9997709468007
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid |     float_x      
-------+------------------
  1000 | -0.8001357298344
  2000 | -0.5870389994234
  3000 |  -0.369545083493
  4000 | -0.1520247617736
  5000 |  0.0664239702746
  6000 |   0.281965653412
  7000 |  0.4925632597879
  8000 |  0.6907451990992
  9000 |  0.9005656959489
 10000 |                 
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | float_x 
-------+---------
  1000 |        
  2000 |        
  3000 |        
  4000 |        
  5000 |        
  6000 |        
  7000 |        
  8000 |        
  9000 |        
 10000 |        
(10 rows)

select rowid,float_x from (select float_x, row_number() over (order by float_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | float_x 
-------+---------
  1000 |        
  2000 |        
  3000 |        
  4000 |        
  5000 |        
  6000 |        
  7000 |        
  8000 |        
  9000 |        
 10000 |        
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid |        nume_x         
-------+-----------------------
  1000 |     0.947302138898522
  2000 |     0.837023418862373
  3000 |      0.72980599058792
  4000 |     0.626219855155796
  5000 |     0.522260365542024
  6000 |      0.41429943870753
  7000 |     0.313118239864707
  8000 |     0.208012379240245
  9000 |     0.105296582449228
 10000 | 0.0000284467823803425
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between     1 and 10000) as t where t.rowid%1000=0;
 rowid |      nume_x       
-------+-------------------
  1000 | 0.105220987461507
  2000 | 0.207979670725763
  3000 | 0.312998778186738
  4000 | 0.413985708728433
  5000 | 0.522132913116366
  6000 | 0.626196458470076
  7000 | 0.729615712538362
  8000 | 0.837006401736289
  9000 |  0.94706692872569
 10000 |                  
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid |       nume_x        
-------+---------------------
  1000 | -0.0572204040363431
  2000 |  -0.157527818344533
  3000 |  -0.262112992350012
  4000 |  -0.366170320659876
  5000 |  -0.469816513359547
  6000 |   -0.57713567186147
  7000 |  -0.682896826416254
  8000 |  -0.785369653720409
  9000 |  -0.893313125707209
 10000 |   -0.99993916740641
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between 10001 and 20000) as t where t.rowid%1000=0;
 rowid |       nume_x        
-------+---------------------
  1000 |  -0.893331774044782
  2000 |  -0.785501023288816
  3000 |  -0.682929904665798
  4000 |  -0.577445950359106
  5000 |  -0.469933776184916
  6000 |  -0.366612220183015
  7000 |  -0.262127089314163
  8000 |  -0.157831937074661
  9000 | -0.0574621357955039
 10000 |                    
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid |       nume_x       
-------+--------------------
  1000 |   0.89450933970511
  2000 |  0.683853192254901
  3000 |   0.47996789496392
  4000 |    0.2625816417858
  5000 | 0.0409852676093578
  6000 | -0.171865842305124
  7000 | -0.384278106503189
  8000 | -0.594578676857054
  9000 | -0.799294468015432
 10000 | -0.999738717451692
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between 20001 and 30000) as t where t.rowid%1000=0;
 rowid |       nume_x       
-------+--------------------
  1000 | -0.799298392608762
  2000 | -0.594680647365749
  3000 | -0.384301328100264
  4000 | -0.172079061158001
  5000 | 0.0407919799908996
  6000 |  0.262404818087816
  7000 |  0.479857482016087
  8000 |  0.683379230089486
  9000 |  0.894455855712295
 10000 |                   
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x desc) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | nume_x 
-------+--------
  1000 |       
  2000 |       
  3000 |       
  4000 |       
  5000 |       
  6000 |       
  7000 |       
  8000 |       
  9000 |       
 10000 |       
(10 rows)

select rowid,nume_x from (select nume_x, row_number() over (order by nume_x asc ) as rowid from strom_test where id between 30001 and 40000) as t where t.rowid%1000=0;
 rowid | nume_x 
-------+--------
  1000 |       
  2000 |       
  3000 |       
  4000 |       
  5000 |       
  6000 |       
  7000 |       
  8000 |       
  9000 |       
 10000 |       
(10 rows)

